packages/aws-cdk-lib/aws-lambda/lib/function-base.ts

```diff
@@ -418,20 +431,29 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
 
   /**
    * Grant the given identity permissions to invoke this Lambda
+   * @param grantee The principal (identity) to grant invocation permission.
+   * @param props onlyGrantLatestVersion (Optional) Controls whether to grant access only to latest function versions. Defaults to `false`.
+   *  - When set to `false`, both the function and functions with specific versions can be invoked.
+   *  - When set to `true`, only the function without a specific version (`$Latest`) can be invoked.
    */
-  public grantInvoke(grantee: iam.IGrantable): iam.Grant {
+  public grantInvoke(grantee: iam.IGrantable, props: GrantInvokeProps = {}): iam.Grant {
     const hash = createHash('sha256')
       .update(JSON.stringify({
         principal: grantee.grantPrincipal.toString(),
         conditions: grantee.grantPrincipal.policyFragment.conditions,
+        onlyGrantLatestVersion: props.onlyGrantLatestVersion,
```

Comment:
Switching between `grantInvoke(grantee)` and `grantInvoke(grantee, { onlyGrantLatestVersion: false })` would cause an unnecessary hash change

```suggestion
        onlyGrantLatestVersion: props.onlyGrantLatestVersion ?? false,
```